File: LeetCode/0792-binary-search/0792-binary-search.py
class Solution:
        def search(self, nums: List[int], target: int) -> int:
            answer = -1 
            start = 0
            end = len(nums) -1
            
            while start <= end:
                mid = (start + end) // 2
                if nums[mid] < target:
                    start = mid + 1
                elif nums[mid] > target:
                    end = mid - 1
                else:
                    answer = mid
                    break
                
            return answer
            

File: LeetCode/0792-binary-search/0792-binary-search.java
class Solution {
    public int search(int[] nums, int target) {
        
        int answer = -1;
        int start = 0;
        int end = nums.length - 1;
        
        while (start <= end) {
            int mid = (start + end) / 2;
            
            if (nums[mid] == target){
                answer = mid;
                break;
            } else if (nums[mid] > target) {
                end = mid - 1;
            } else {
                start = mid + 1;
            }
        }
        
        return answer;
    }
}